ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -220,442 +223,447 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
-        filter(address => !this.enoughDataToVerify(address)),
-        tap(() => addressSuggestion$.next(null)),
-        map(address => [address, this.computeValidityForSparseAddress(address)]),
-        mergeMap(([address, isValid]) => {
+    private initializeComponentState(): void {
+        const initialState: ComponentState = {
+            isReadOnly: false,
+            activityInstanceGuid: null,
+            showSuggestion: false,
+            enteredAddress: null,
+            suggestedAddress: null,
+            formWarningMessages: [],
+            formErrorMessages: [],
+            fieldErrors: [],
+            isTemporarilyDisabled: false
+        };
+
+        this.state$ = this.stateUpdates$.pipe(
+            startWith(initialState),
+            scan((acc: ComponentState, update) => ({...acc, ...update})),
+            tap(state => this.logger.logDebug(`${this.LOG_SOURCE}. New embeddedComponentState$=%o`, state)),
+            shareReplay(1)
+        );
+        this.stateSubscription = this.state$.pipe(
+            takeUntil(this.ngUnsubscribe))
+            .subscribe();
+    }
+
+    ngOnInit(): void {
+        this.setupActions();
+    }
+
+    setupActions(): void {
+        const verificationError$ = new Subject<any>();
+        const addressSuggestion$ = new Subject<AddressSuggestion | null>();
+        const busyCounter$ = new BehaviorSubject(0);
+        const isThisComponentBusy$ = busyCounter$.pipe(
+            scan((acc, val) => acc + val, 0),
+            map(val => val > 0),
+            distinctUntilChanged()
+        );
+
+        const initializeStateAction$ = this.state$.pipe(
+            take(1),
+            tap(() => busyCounter$.next(1)),
+            // let's disable form while we try to load initial address
+            tap(() => this.stateUpdates$.next({isTemporarilyDisabled: true})),
+            mergeMap((state) => this.addressService.findDefaultAddress().pipe(
+                catchError(error => {
+                    this.logger.logWarning(this.LOG_SOURCE, `An error occurred during findDefaultAddress: ${error.code}, ${error.message}`);
+                    return of(null);
+                }),
+                map(defaultAddress => [state, defaultAddress]),
+            )),
+            tap(([_, defaultAddress]: [ComponentState, Address | null]) =>
+                defaultAddress && this.inputAddress$.next(defaultAddress)),
+            take(1),
+            // filter for case where we need to go on to look for a temp address?
+            filter(([state, defaultAddress]) => !defaultAddress && !!(state as ComponentState).activityInstanceGuid),
+            // map(([state, _]) => state as ComponentState),
+            mergeMap(([state, _]) => this.addressService.getTempAddress(state.activityInstanceGuid).pipe(
+                catchError(error => {
+                    this.logger.logWarning(this.LOG_SOURCE, `An error occurred during getTempAddress: ${error.code}, ${error.message}`);
```

Comment:
The question:
should we report about these errors (no default/temp address) to GCP  
or it is just enough to write a warning to console ?

Reply:
In these 2 cases here, I think the right think to do is to capture (like you are doing), and then log as as debug (logDebug , I believe)?
I think our base code that does the http calls might be a little too creative regarding capturing certain errors, and a bunch of other stuff, but regardless the two "not founds" are not really exceptions; client just checks if an address exists and the answer, sometimes is, it does not (404).